core/block_validator_test.go

```diff
@@ -138,7 +138,7 @@ func testHeaderVerificationForMerging(t *testing.T, isClique bool) {
 
 		td := 0
 		genDb, blocks, _ := GenerateChainWithGenesis(gspec, engine, 8, nil)
-		for _, block := range preBlocks {
+		for _, block := range blocks {
```

Comment:
The previous code is _wrong_, since `preBlocks` is empty. The big quesion is not whether the new code is correct, but rather "why did this not cause some failure earlier" -- maybe we need to first fix the test so that it fails properly, before we fix this flaw?

Reply:
I found the issue because we have a modification to `IsTTDReached ` which short circuits to `true` if `TerminalTotalDifficulty == 0`. I'll look into why this test isn't failing.